keps/sig-network/4858-ip-cidr-validation/README.md

```diff
@@ -0,0 +1,838 @@
+
+- `k8s.io/apimachinery/pkg/util/validation`: `2024-10-29` - `82.3`
+- `k8s.io/kubernetes/pkg/api/pod/warnings.go`: `2024-10-29` - `97.4`
+- `k8s.io/kubernetes/pkg/api/service/warnings.go`: `2024-10-29` - `96.8`
+- `k8s.io/kubernetes/pkg/apis/core/validation`: `2024-10-29` - `84.4`
+- `k8s.io/kubernetes/pkg/apis/discovery/validation`: `2024-10-29` - `98.8`
+- `k8s.io/kubernetes/pkg/apis/networking/validation`: `2024-10-29` - `91.4`
+- `k8s.io/kubernetes/pkg/registry/core/endpoint`: `2024-10-29` - `0.0`
+- `k8s.io/kubernetes/pkg/registry/core/node`: `2024-10-29` - `50.0`
+- `k8s.io/kubernetes/pkg/registry/core/pod`: `2024-10-29` - `64.0`
+- `k8s.io/kubernetes/pkg/registry/core/service`: `2024-10-29` - `73.8`
+- `k8s.io/kubernetes/pkg/registry/discovery/endpointslice`: `2024-10-29` - `79.2`
+- `k8s.io/kubernetes/pkg/registry/networking/ingress`: `2024-10-29` - `64.5`
+- `k8s.io/kubernetes/pkg/registry/networking/networkpolicy`: `2024-10-29` - `76.9`
+
+##### Integration tests
+
+No new tests, and we will remove
+`test/integration/apiserver/cve_2021_29923_test.go`, which tests that
+it is possible to create new objects with invalid IP values (since it
+won't be possible any more).
+
+##### e2e tests
+
+No new tests, and we will remove `test/e2e/network/funny_ips.go`,
+which tests the behavior of kube-proxy when it sees
+Service/EndpointSlice objects with invalid IPs (since it won't be
+possible to create such objects any more). We already added [equivalent
+unit tests] to the iptables and nftables backends of kube-proxy (which
+actually caught a bug in kube-proxy that the e2e test hadn't).
+
+This means that we will lose testing for out-of-tree service proxy
+implementations (and out-of-tree NetworkPolicy implementations, etc),
+but there doesn't seem to be anything we can do about this: we can
+only have an e2e test of "what does the implementation do with bad IP
+values?" if we can create Pods / Services / EndpointSlices /
+NetworkPolicies / etc with bad IP values. But the whole point of this
+KEP is to make that impossible.
+
+(I guess possibly we might be able to set up the e2e test to be able
+to write the object directly into etcd, bypassing the apiserver? Or
+maybe add an "e2e test run" mode to the apiserver, that makes it
+possible to request that it skip validation for a particular
+Create/Update? Alternatively, this is an argument for eventually
+forcibly fixing all remaining invalid values in etcd, since it will no
+longer be possible to test behavior regarding them.)
+
+[equivalent unit tests]: https://github.com/kubernetes/kubernetes/pull/126203
+
+### Graduation Criteria
+
+#### Alpha
+
+- API warnings added.
+
+- Mandatory strict validation for `NetworkDeviceData`.
+
+- All kube-internal controllers only write out canonical IPs/CIDRs.
+
+- Updated validation available behind a feature gate, but disabled by
+  default.
+
+- Unit tests added/updated.
+
+#### Beta
+
+- Updated validation available behind a feature gate, enabled by
```

Comment:
At minimum I'd expect all unresolved items be resolved before entering beta. Also for GA, that's where I'd add the integration removal.